[dynamo] support tracing new typing union syntax X | Y

williamwen42 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #166599

To do in a followup - I think there's an approach to reconstruct typing variables.

cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @pragupta @msaroufim @dcci @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @Lucaskabela

williamwen42 committed:
[dynamo] support tracing new typing union syntax X | Y

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo] support tracing new typing union syntax X | Y"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

SherlockNoMad commented:
feel free to remove the test case added in https://github.com/pytorch/pytorch/pull/166443

Skylion007 commented on torch/_dynamo/variables/builder.py:
```diff
@@ -3745,7 +3745,9 @@ def create(tx: "InstructionTranslator", value) -> VariableTracker:
-        elif isinstance(value, types.GenericAlias):
+        elif isinstance(value, types.GenericAlias) or isinstance(
+            value, types.UnionType
+        ):
```
```suggestion
        elif isinstance(value, (types.GenericAlias, types.UnionType)):
```

williamwen42 committed:
Update on "[dynamo] support tracing new typing union syntax X | Y"


To do in a followup - I think there's an approach to reconstruct typing variables.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo] support tracing new typing union syntax X | Y"


To do in a followup - I think there's an approach to reconstruct typing variables.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

williamwen42 commented:
@SherlockNoMad updated the test you added - can you review that?
